flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/codegen/calls/FunctionGenerator.scala

```diff
@@ -801,6 +801,20 @@ object FunctionGenerator {
       new MethodCallGen(
         BuiltInMethod.IS_JSON_SCALAR.method)))
 
+  addSqlFunctionMethod(
+    JSON_EXISTS,
+    Seq(CHAR, CHAR),
```

Comment:
What about `Seq(VARCHAR, CHAR)`/`Seq(CHAR, VARCHAR)`?

Reply:
> What about `Seq(VARCHAR, CHAR)`/`Seq(CHAR, VARCHAR)`?

```
  public SqlJsonExistsFunction() {
    super("JSON_EXISTS", SqlKind.OTHER_FUNCTION,
        ReturnTypes.cascade(ReturnTypes.BOOLEAN, SqlTypeTransforms.FORCE_NULLABLE), null,
        OperandTypes.or(
            OperandTypes.family(SqlTypeFamily.ANY, SqlTypeFamily.CHARACTER),
            OperandTypes.family(SqlTypeFamily.ANY, SqlTypeFamily.CHARACTER, SqlTypeFamily.ANY)),
        SqlFunctionCategory.SYSTEM);
  }
```
The `path` parameter can only be CHAR.  This need to add `Seq(VARCHAR, CHAR)`

Reply:
According to SQL 2016 Standard: 
```
<JSON API common syntax> ::= <JSON context item> <comma> <JSON path specification> [ AS <JSON table path name> ] [ <JSON passing clause> ]
<JSON context item> ::= <JSON value expression>
<JSON path specification> ::= <character string literal>
```
Second parameter can only be string literal, and in Flink, it's `CHAR` type.

Reply:
> According to SQL 2016 Standard:
> 
> ```
> <JSON API common syntax> ::= <JSON context item> <comma> <JSON path specification> [ AS <JSON table path name> ] [ <JSON passing clause> ]
> <JSON context item> ::= <JSON value expression>
> <JSON path specification> ::= <character string literal>
> ```
> 
> Second parameter can only be string literal, and in Flink, it's `CHAR` type.

Thank you very much for your analysis, it has helped me a lot.